decentdao/decent-interface
# `[roles-0.2.0]` Claim hat stream

This PR implements basic UI and logic for withdrawing payment to the hat wearer.
It also fixes filtering of added/edited payments as previously added payment was also considered as edited.

Closes #2186 

Testing steps:

1. Create payment for any given role, where you're a wearer. Make total length of the payment short enough so that you could easily withdraw sufficient amount of tokens (1 hour for 1000 tokens should be good).
2. Vote and execute proposal for creating payment to the role
3. Go to the role details - payment info should appear there
4. Notice "Withdraw" section appearing in the payment details
5. Enter amount to withdraw (do not put maximum amount just yet) and hit "Withdraw" - this will prompt to sign transaction
6. Make sure proper amount of tokens were sent to your wallet
7. Now, hit "max" button and hit "Withdraw" and sign transaction
8. Make sure proper amount of all available for withdrawing tokens were sent to your wallet

Basic design was grabbed from https://www.figma.com/design/9hUYtvCKjOyB92vt8ukYtD/Product-Handoff?node-id=4087-18066&m=dev but not implemented 1-1, since that design should change (afaik)
<img width="1172" alt="Screenshot 2024-08-02 at 16 33 30" src="https://github.com/user-attachments/assets/fc38054c-af8c-4260-9b95-580fd3c0de90">
